Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/CalibrationAPI.cs

```diff
@@ -0,0 +1,389 @@
+                    var calcExtrinsics = new CalculatedCameraExtrinsics();
+                    calcExtrinsics.ViewFromWorld = Matrix4x4.TRS(inversePosition, inverseRotation, Vector3.one);
+                    calcExtrinsics.Succeeded = succeeded;
+                    output.Add(calcExtrinsics);
+                }
+
+                return output;
+            }
+        }
+
+        public CalculatedCameraExtrinsics CalculateGlobalArUcoExtrinsics(CameraIntrinsics intrinsics)
+        {
+            int intrinsicsSize = 12;
+            float[] inputIntrinsics = new float[intrinsicsSize];
+            inputIntrinsics[0] = intrinsics.FocalLength.x;
+            inputIntrinsics[1] = intrinsics.FocalLength.y;
+            inputIntrinsics[2] = intrinsics.PrincipalPoint.x;
+            inputIntrinsics[3] = intrinsics.PrincipalPoint.y;
+            inputIntrinsics[4] = intrinsics.RadialDistortion.x;
+            inputIntrinsics[5] = intrinsics.RadialDistortion.y;
+            inputIntrinsics[6] = intrinsics.RadialDistortion.z;
+            inputIntrinsics[7] = intrinsics.TangentialDistortion.x;
+            inputIntrinsics[8] = intrinsics.TangentialDistortion.y;
+            inputIntrinsics[9] = intrinsics.ImageWidth;
+            inputIntrinsics[10] = intrinsics.ImageHeight;
+            inputIntrinsics[11] = 0.0f; // reprojection error, unused
+
+            int sizeExtrinsics = 7;
+            int numExtrinsics = 1;
+            float[] extrinsics = new float[sizeExtrinsics];
+            if (!ProcessGlobalArUcoExtrinsicsNative(inputIntrinsics, extrinsics, numExtrinsics))
+            {
+                PrintLastError();
+                return null;
+            }
+
+            bool succeeded = (extrinsics[0] > 0.01f);
+
+            Vector3 rightHandedPosition = new Vector3(extrinsics[1], extrinsics[2], extrinsics[3]);
+
+            Vector3 rodriguesVector = new Vector3(extrinsics[4], extrinsics[5], extrinsics[6]);
+            var angle = Mathf.Rad2Deg * rodriguesVector.magnitude;
+            var axis = rodriguesVector.normalized;
+            Quaternion rightHandedRotation = Quaternion.AngleAxis(angle, axis);
+
+            var rightHandedMatrix = Matrix4x4.TRS(rightHandedPosition, rightHandedRotation, Vector3.one);
+            var leftHandedMatrix = rightHandedMatrix;
+            leftHandedMatrix.m02 *= -1.0f;
+            leftHandedMatrix.m12 *= -1.0f;
+            leftHandedMatrix.m22 *= -1.0f;
+            leftHandedMatrix.m32 *= -1.0f;
+
+            var inverse = leftHandedMatrix.inverse;
+            var inversePosition = inverse.GetColumn(3);
+            var inverseRotation = Quaternion.LookRotation(inverse.GetColumn(2), inverse.GetColumn(1)) * Quaternion.Euler(0, 0, 180);
+
+            var calcExtrinsics = new CalculatedCameraExtrinsics();
+            calcExtrinsics.ViewFromWorld = Matrix4x4.TRS(inversePosition, inverseRotation, Vector3.one);
+            calcExtrinsics.Succeeded = succeeded;
+            return calcExtrinsics;
```

Comment:
This section also has some copiage from the function above, consider making some of this common as well.